docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,128 @@
+= Git Submodule guideline
+
+This document outlines some basic guidelines for synchronizing changes between an upstream and downstream repository using Git submodules.
+
+By leveraging Git submodules, it is possible to maintain a consistent and up-to-date relationship between related repositories (upstream and downstream) while having more control of how the project is built, tested and released.
+
+Additionally, the integration with Konflux can automate key aspects of this process, ensuring that changes from the upstream repository are efficiently and reliably propagated to the downstream repository.
+
+== Implementing this guideline
+=== Basic definitions
+
+- **Upstream repo**: a community project which you do not have push access to (or which you do not want to push CI changes into).
+
+- **Downstream repo**: a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+=== Create a downstream repo
+
+The downstream repo can be used to do the following:
+
+* hold the git submodule reference to the upstream repository.
+* use the contents of the git submodule in a Containerfile for further edits.
+* onboarded onto {ProductName}.
+
+==== Configuring components
+
+Here is an example of a component definition:
+
+[source, yaml]
+----
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Component
+metadata:
+  name: <component-name>
+  namespace: <project-namespace>
+  annotations:
+    build.appstudio.openshift.io/request: configure-pac
+    image.redhat.com/generate: '{"visibility": "public"}'
+spec:
+  application: <application-name>
+  componentName: <component-name>
+  source:
+    git:
+      revision: main
+      url: <downstream repo name>
+      dockerfileUrl: <Dockerfile/Containefile path in downstream repository>
+      context: <directory path used to build the Containerfile/Dockerfile>
+----
+
+See xref:/how-tos/creating.adoc[Creating applications and components] for more details.
+
+==== Configuring the Containerfile
+
+In the Containerfile you can specify the git submodule directories and add any necessary edits that differ from the upstream repository. Here is a basic example:
+
+[source, Dockerfile]
+----
+FROM <base-image>
+ENV <SUBMODULEPKG> /path/to/submodule
+
+# Install necessary dependencies
+RUN dnf -y install <dependencies>
+
+# Set the working directory
+WORKDIR ${<SUBMODULEPKG>}
+
+RUN sed -i 's/<old-text>/<new-text>/g' /file/in/submodule/directory
```

Comment:
You only show changing text here. Would it be better to show content that is more generic like

```dockerfile
# Commands to update the submodule for the current image build
RUN sed -i 's/<old-text>/<new-text>/g' file/in/submodule/directory
# Commands to build binary from submodule sources
RUN go build file/in/submodule/directory
```

(Also, removing the leading `/` would mean that you are working within the submodule working directory instead of needing to pass the absolute path).

Reply:
Added, thanks! -> https://github.com/konflux-ci/docs/pull/120/commits/f1be568ae08b9605e685c3d1e1e288862518855a#diff-46f8b5b0ae481a31c84644c1fde822df6be41954ac5d4002de9444fe1b91fd08R43-R46